File: app/src/main/java/org/bibletranslationtools/sun/ui/activities/auth/AuthenticationActivity.java
package org.bibletranslationtools.sun.ui.activities.auth;

import android.content.Intent;
import android.view.View;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import org.bibletranslationtools.sun.adapter.viewpager.OnboardingAdapter;
import org.bibletranslationtools.sun.databinding.ActivityAuthenticationBinding;
import org.bibletranslationtools.sun.preferen.UserSharePreferences;
import org.bibletranslationtools.sun.ui.activities.MainActivity;

public class AuthenticationActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        UserSharePreferences userSharePreferences = new UserSharePreferences(this);

        // Assuming that userSharePreferences is initialized somewhere else
//        if (userSharePreferences.getLogin()) {
//            startActivity(new Intent(this, MainActivity.class));
//            finish();
//        }

        // Inflate the layout
        ActivityAuthenticationBinding binding = ActivityAuthenticationBinding.inflate(getLayoutInflater());
        final View view = binding.getRoot();
        setContentView(view);

        // Setup onboarding
        OnboardingAdapter onboardingAdapter = new OnboardingAdapter(this);
        binding.onboardingVp.setAdapter(onboardingAdapter);
        binding.indicator.setViewPager(binding.onboardingVp);

        // Setup view now button
        binding.startNowBtn.setOnClickListener(v -> {
            startActivity(new Intent(this, MainActivity.class));
            finish();
        });

    }
}

File: app/src/main/kotlin/org/bibletranslationtools/sun/ui/activities/auth/AuthenticationActivity.kt
package org.bibletranslationtools.sun.ui.activities.auth

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import org.bibletranslationtools.sun.adapter.viewpager.OnboardingAdapter
import org.bibletranslationtools.sun.databinding.ActivityAuthenticationBinding
import org.bibletranslationtools.sun.preferences.UserSharePreferences
import org.bibletranslationtools.sun.ui.activities.MainActivity

class AuthenticationActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val userSharePreferences = UserSharePreferences(this)

        // Inflate the layout
        val binding = ActivityAuthenticationBinding.inflate(layoutInflater)
        setContentView(binding.root)


        // Setup onboarding
        val onboardingAdapter = OnboardingAdapter(this)
        binding.onboardingVp.adapter = onboardingAdapter
        binding.indicator.setViewPager(binding.onboardingVp)


        // Setup view now button
        binding.startNowBtn.setOnClickListener { v ->
            startActivity(Intent(this, MainActivity::class.java))
            finish()
        }
    }
}